flytekit/extras/sqlalchemy/task.py

```diff
@@ -0,0 +1,84 @@
+import typing
+from dataclasses import dataclass
+
+import pandas as pd
+from sqlalchemy import create_engine
+
+from flytekit import current_context, kwtypes
+from flytekit.core.base_sql_task import SQLTask
+from flytekit.core.python_function_task import PythonInstanceTask
+from flytekit.types.schema import FlyteSchema
+
+
+@dataclass
+class SQLAlchemyConfig(object):
+    """
+    Use this configuration to configure task. String should be standard
+    sqlalchemy connector format
+    (https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls).
+    Database can be found:
+      - within the container
+      - or from a publicly accessible source
+
+    Args:
+        uri: default sqlalchemy connector
+        connect_args: sqlalchemy kwarg overrides -- ex: host
+        secret_connect_args: flyte secrets loaded into sqlalchemy connect args
+            -- ex: {"password": {"name": SECRET_NAME, "group": SECRET_GROUP}}
+    """
+
+    uri: str
+    connect_args: typing.Optional[typing.Dict[str, typing.Any]] = None
+    secret_connect_args: typing.Optional[typing.Dict[str, typing.Dict[str, typing.Any]]] = None
+
+
+class SQLAlchemyTask(PythonInstanceTask[SQLAlchemyConfig], SQLTask[SQLAlchemyConfig]):
+    """
+    Makes it possible to run client side SQLAlchemy queries that optionally return a FlyteSchema object
+
+    TODO: How should we use pre-built containers for running portable tasks like this. Should this always be a
+          referenced task type?
+    """
+
+    _SQLALCHEMY_TASK_TYPE = "sqlalchemy"
+
+    def __init__(
+        self,
+        name: str,
+        query_template: str,
+        task_config: SQLAlchemyConfig,
+        inputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
+        output_schema_type: typing.Optional[typing.Type[FlyteSchema]] = None,
+        **kwargs,
+    ):
+        outputs = kwtypes(results=output_schema_type if output_schema_type else FlyteSchema)
+        self._uri = task_config.uri
+        self._connect_args = task_config.connect_args or {}
+        self._secret_connect_args = task_config.secret_connect_args
```

Comment:
I think we should patch the secrets from here into the parent constructor and then we can easily drop **kwargs.
Let me know if you want to chat about this, and maybe the model we have needs a little bit of an update to clean it up

Reply:
if we're using `secret_requests` for config overrides we'll still need the key mapping somehow, unless we assume one of the Secret fields is the key